lib/room.rb

```diff
@@ -0,0 +1,86 @@
+module Hotel
+  class Room
+    attr_reader :room_id
+
+    RATE = 200.freeze
+    private_constant :RATE
+
+    def initialize(room_id)
+      @room_id = room_id
+      @reservations = Array.new
+      @block_intervals = Array.new
+    end
+
+    def reserve(reserved_dates)
+      if !is_available?(reserved_dates)
+        raise ArgumentError, "The room you want to reserve is not available."
+      end
+
+      r = Hotel::Reservation.new(reserved_dates, @room_id, RATE)
```

Comment:
Ruby will be looking for names locally first, prioritizing local names, since they're all in the same module, there's no need to namespace `Hotel::Reservation`. `Reservation` should just work! Same with the rest of the namespacing in the code.